cloud_images/centos7/dcos_vol_setup.sh

```diff
@@ -53,7 +53,12 @@ function main {
     echo "Adding entry to fstab: $fstab"
     echo "$fstab" >> /etc/fstab
     echo -n "Mounting: $device to $mount_location"
-    mount "$mount_location"
+    until grep ^$device /etc/mtab > /dev/null; do sleep 1; echo -n .; mount "$mount_location" || :; done
+    if [ "$mount_location" = "/var/log" ]; then
+      echo "Recreating /var/log/journal"
+      mkdir /var/log/journal
+      systemd-tmpfiles --create --prefix /var/log/journal
+    fi
```

Comment:
The journal was preserved when I tested this change, but perhaps it was because journald was eventually restarted by cloud-final (long after docs-vol-setup completed).

I'm trying a journald restart from within the script with a handler installed. Stand by.